### PR TITLE
Notification hub

### PR DESCRIPTION
**Pull request by senechal**

### CODE REVIEW

File: raccoon_services/services/notification_helper.py [hunk 1]
```diff
@@ -1,58 +1,176 @@
...
+
+    def clear_sender(self):
+        self.data = dict(software_name=self.software_name)
+
+    def _validate_data(self):
+        return self.data != dict(software_name=self.software_name)
+
+    def add_schedule_time(self, timestamp):
+        self.data.update({
+            "timestamp": self._validate_schedule(timestamp)
```
Comment (arielmorelli) [inline raccoon_services/services/notification_helper.py]: Not following the pattern: dict(KEY=value) [lines :18 and :21] and using { key: value}

File: raccoon_services/services/notification_helper.py [hunk 2]
```diff
@@ -1,58 +1,176 @@
...
-        if phone_list:
-            valid_phones = [self._validate_phone_number(x) for x in phone_list]
-            if len(valid_phones) >= 1:
-                url = self.url + '/notification'
-                payload = self._create_json(phones=valid_phones,
-                                            message=message, schedule=schedule)
+    def _validate_url(self, url):
+        if url:
+            parsed = urlparse(url)
+            if len(parsed.scheme) > 0 and len(parsed.netloc) > 0:
```
Comment (arielmorelli) [inline raccoon_services/services/notification_helper.py]: if len(parsed.scheme) ~> 0~ and len(parsed.netloc) ~> 0~:

File: raccoon_services/services/notification_helper.py [hunk 3]
```diff
@@ -1,58 +1,176 @@
...
+                return url
+            else:
+                raise Exception("Invalid URL")
```
Comment (arielmorelli) [inline raccoon_services/services/notification_helper.py]: Maybe:
```
raise Exception("Invalid URL ({})".format(url))
```

File: raccoon_services/services/notification_helper.py (hunk 2 above)
Comment (senechal) [inline raccoon_services/services/notification_helper.py]: Dont even look this method, just move it to that place.

**Pull request merged**